plugin-build/src/test/kotlin/io/sentry/android/gradle/SentryTaskProviderTest.kt

```diff
@@ -0,0 +1,164 @@
+package io.sentry.android.gradle
+
+import com.android.build.gradle.AppExtension
+import io.sentry.android.gradle.SentryTasksProvider.getAssembleTask
+import io.sentry.android.gradle.SentryTasksProvider.getBundleTask
+import io.sentry.android.gradle.SentryTasksProvider.getDexTask
+import io.sentry.android.gradle.SentryTasksProvider.getPackageTask
+import io.sentry.android.gradle.SentryTasksProvider.getPreBundleTask
+import io.sentry.android.gradle.SentryTasksProvider.getTransformerTask
+import org.gradle.testfixtures.ProjectBuilder
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+
+class SentryTaskProviderTest {
+
+    @get:Rule
+    val testProjectDir = TemporaryFolder()
+
+    @Test
+    fun `getTransformerTask returns null for missing task`() {
+        val project = ProjectBuilder.builder().build()
+
+        assertNull(getTransformerTask(project, "debug"))
+    }
+
+    @Test
+    fun `getTransformerTask returns transform for R8`() {
+        val project = ProjectBuilder.builder().build()
+        val task = project.tasks.register("transformClassesAndResourcesWithR8ForDebug")
```

Comment:
these 2 lines are repeated across all the tests, maybe let's add a method to not DRY?

`registerTask(name: String): Tuple<Project, Task>` or something like that

what do you think?

Reply:
Yup added 👍